paket.dependencies

```diff
@@ -1,35 +1,59 @@
 source http://nuget.org/api/v2
+redirects: on
+strategy: max
 
-nuget FSharp.Compiler.Service
 nuget Fantomas
+nuget FSharp.Compiler.Service
+nuget FSharp.Core
+nuget FSharp.Management
 nuget FSharp.ViewModule.Core
-nuget FsXaml.Wpf
-nuget SourceLink.SymbolStore 0.5.0
 nuget FSharpLint.Core prerelease
-nuget VSSDK.IDE.Deploy.14
-nuget FSharp.Management
 nuget FsPickler
-nuget FSharp.Core redirects:on
+nuget FsXaml.Wpf
+nuget SourceLink.SymbolStore 0.5.0
 
+group vssdklegacy
+    source http://nuget.org/api/v2
+    framework >= net40
+    
+    nuget Microsoft.VisualStudio.Shell.Interop.8.0
+    nuget Microsoft.VisualStudio.Shell.Immutable.10.0
+    nuget VSSDK.ComponentModelHost.11
+    nuget VSSDK.DTE
+    nuget VSSDK.DTE.8
+    nuget VSSDK.DTE.10
+    nuget VSSDK.NavigateTo
+    nuget VSSDK.VSLangProj
+    
+    
+group vssdk
+    source http://nuget.org/api/v2
+    framework >= net45
+    
+    nuget Microsoft.VisualStudio.Editor
+    nuget Microsoft.VisualStudio.Language.Intellisense
```

Comment:
> Again sorry about my Paket ignorance here. But doesn't the lock file represent the result of a restore vs. specifying what packages are needed?

no changes to paket.lock would only happen on `paket upgrade` or `paket install`, normal workflow (for someone building) invokes `paket restore` which will strictly abide to the `paket.lock` file (or fail if packages aren't available).

> Basically I'm interested in what mechanism would stop the packages from silently upgrading to 15.0.

the truth is in `paket.lock` and this file isn't touched unless someone invokes the `install` or `upgrade` commands, we can also specify `< 15` in `paket.dependencies` as well which will in turn avoid `install` or `upgrade` to look at eventual versions 15 or above.

> or values returned from the SDK.

Is it related to GAC? Is there any way to check that value?

>  I believe HintPath does take precedence today but I wouldn't bet on it, definitely wouldn't bet on it being true forever.

Is there a way to specify file reference in a more deterministic fashion? (fails if file not at location, no fallback).

> This is part of the reason I eventually switched to always using strongly named references for the VS editor binaries. It allowed me to take MSBuild resolution out of the equation. No matter what the rules are it will only succeed if it matches the exact assembly that I want :smile:

This is interesting, although it adds a lot of noise to the *proj files, also I think paket doesn't ever write the version in *proj files, I assume main reasons being:

* all versions are listed in sole `paket.lock` no need for redundant (and possibly mismatching) information
* code churn on project files
* assumption that assembly file reference would be determining which assembly ends up being referenced